crates/gpui/src/text_system/font_features.rs

```diff
@@ -7,10 +11,14 @@ macro_rules! create_definitions {
     ($($(#[$meta:meta])* ($name:ident, $idx:expr)),* $(,)?) => {
 
         /// The OpenType features that can be configured for a given font.
-        #[derive(Default, Copy, Clone, Eq, PartialEq, Hash)]
+        #[derive(Default, Clone, Eq, PartialEq, Hash)]
         pub struct FontFeatures {
             enabled: u64,
             disabled: u64,
+            #[cfg(target_os = "windows")]
+            other_enabled: SharedString,
```

Comment:
Why are these additional fields needed?

Reply:
The `other_enabled` and `other_disabled` serve the same function as `Vec<FeatureName>`. Currently, Zed only supports 34 OpenType features from `calt` to `zero`, stored in `enabled: u64` and `disabled: u64` as bitfields, allowing for a maximum of 64 features. In theory, there can be up to 36*36*36*36 OpenType features. For instance, `Firacode` supports features like `cv01` to `cv30`, while some special fonts may even use `cv99` for certain reasons. Therefore, to support all OpenType features, the current `FontFeatures` struct would need to introduce a structure similar to `Vec<T>`, which would not support `Copy`. Cloning operations with `Vec<T>` are relatively costly, so `SharedString` is used here since its clone operation simply duplicates the pointer.

This change might have some impact on the performance of macOS, but I assume it shouldn't be significant. In macOS, `FontFeatures` has only two members of type `u64`, so the performance gap between clone and copy should be minimal. Alternatively, I could separate the implementation of `FontFeatures` for macOS and Windows without altering the existing macOS code.

Reply:
If this is a correctness issue, e.g. both implementation are wrong, then we shouldn't be hiding this behind a `#[cfg]` and should enable it for everything. As for an efficient representation, I think this packed representation makes sense. Maybe let's document the other_enabled / other_disabled format (e.g. 'a list of 4 character font feature values'). I think I'd prefer something like a `ArcCow<'static, [char; 4]>`, so that it's super explicit what's being stored, but I expect that's a lot more trouble to work with.

That said, I expect the common case is that there are no additional font features? For that case, I think it would be nice to special case the [into call on 165](https://github.com/zed-industries/zed/pull/10756/files#diff-5feedf70b6351460e162111e1a1966afb950f03ecee20d6eaa6aaa35b28a12acR165). If the strings are empty, we can call `"".into()` and get a static reference copy, which should be even cheaper than the Arc clone.

Reply:
> If this is a correctness issue, where both implementation are wrong, then we shouldn't be hiding this behind a `#[cfg]` and should enable it for everything. As for an efficient representation, I think this packed representation makes sense. Maybe let's document the other_enabled / other_disabled format (e.g. 'a list of 4 character font feature values'). I think I'd want to use something like a `ArcCow<'static, [char; 4]>`, so that it's super explicit what's being stored, but I expect that's a lot more trouble to work with and it will be even less efficient than a `str` representation, as the initial ascii characters won't be packed into a single u8.

I'm sorry, I didn't fully follow you. Here, it's not just a correctness issue; currently, Zed only supports a subset of [font features](https://github.com/zed-industries/zed/blob/main/crates/gpui/src/text_system/font_features.rs#L155-#L190). The font features that Zed doesn't support but are supported by DirectWrite (`cv99` for example) are stored here.

Assuming the user has the following settings:
```rust
  "buffer_font_features": {
      "ss01": true,    <-- stored in `enabled` as bitfield
      "ss02": true,    ...
      "ss03": true,    ...
      "ss04": true,    ...
      "ss05": true,    ...
      "cv01": true,    <-- stored in `other_enabled`
      "cv02": true,
      "cv03": true,
      "cv04": true,
    }
```

The `FontFeature`:
```rust
other_enabled: "cv01cv02cv03cv04"
other_disabled: ""
```

> That said, I expect the common case is that there are no additional font features? For that case, I think it would be nice to special case the [into call on 165](https://github.com/zed-industries/zed/pull/10756/files#diff-5feedf70b6351460e162111e1a1966afb950f03ecee20d6eaa6aaa35b28a12acR165). If the strings are empty, we can call `"".into()` to get a SharedString wrapping a static reference. Cloning that should be even cheaper than the Arc clone, and should mean there's next to no performance hit.

This is a nice idea.

Reply:
What I mean is:
 - can you have more than 34 open type features on macOS or Linux?
 - If yes, then we should enable this for all platforms
 - If no, then we should leave this enabled just for windows.
 
Looking at [wikipedia](https://en.wikipedia.org/wiki/List_of_typographic_features), it seems likely that OpenType can have more than 34 features on all platforms. As such, we should enable this change for all platforms.

Reply:
My idea is that the current PR only provides implementations for additional font features on the `Windows` platform, without addressing other platforms. Therefore, in this PR, minimize the impact, I only enabling it for Windows for now. If other platforms implement support in the future, we can then remove the corresponding `cfg` tags. 

To be honest, Linux should be able to support other font features, but I'm unsure about macOS. On macOS, `Core Text` uses a `selector` mechanism to enable specific font features. However, as I mentioned, theoretically, all 26 English letters and 10 numbers could be considered "valid" OpenType font features. For example, the [Iosevka](https://github.com/be5invis/Iosevka/blob/main/doc/character-variants.md) font uses `VS**` and `VL**`, which are not listed in [wiki](https://en.wikipedia.org/wiki/List_of_typographic_features). It's challenging to toggle these font features on macOS.

Reply:
I see, so you're saying it takes more work to fully utilize these on other platforms? That makes sense then. I'll merge this then :)

Reply:
> I see, so you're saying it takes more work to fully utilize these on other platforms? That makes sense then. I'll merge this then :)

Especially on macOS, I couldn't find the `selector` for toggling these `cv**` font features like `cv01`, but almost every font is utilizing `cv**`.